setup.sh

```diff
@@ -41,59 +41,60 @@ declare -r DOCUMENTATION='documentation'
 
 declare -r CONFIGS_PATH='configs'
 
+debian_pkg_check()
+{
+  dpkg-query -W --showformat='${Status}\n' "$1" 2> /dev/null | grep -c 'ok installed'
+}
+
 function check_dependencies()
 {
   local package_list=''
   local pip_package_list=''
   local cmd=''
   local distro
-
+  declare -a debian=('debian_pkg_check ' 'apt install -y')
+  declare -a arch=('pacman -Ql ' 'pacman -S')
+  declare -a gentoo=('qlist -I ' 'emerge')
+  declare -a fedora=('rpm -q ' 'dnf install -y')
+  declare -A distro_commands=(['arch']=arch ['gentoo']=gentoo ['fedora']=fedora ['debian']=debian)
+  declare -n distro_cmd
   distro=$(detect_distro '/')
 
-  if [[ "$distro" =~ 'arch' ]]; then
-    while IFS='' read -r package; do
-      installed=$(pacman -Ql "$package" &> /dev/null)
-      [[ "$?" != 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/arch.dependencies"
-    cmd="pacman -S $package_list"
-  elif [[ "$distro" =~ 'debian' ]]; then
-    while IFS='' read -r package; do
-      installed=$(dpkg-query -W --showformat='${Status}\n' "$package" 2> /dev/null | grep -c 'ok installed')
-      [[ "$installed" -eq 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/debian.dependencies"
-    cmd="apt install -y $package_list"
-  elif [[ "$distro" =~ 'fedora' ]]; then
+  distro_cmd="${distro_commands["$distro"]}"
+  distro_package_manager_cmd="${distro_cmd[1]}"
+  distro_package_check="${distro_cmd[0]}"
```

Comment:
To make the code more readable, you can create a global readonly variable that maps 0 to something like 'IS_INSTALLED_CMD' and 1 to `DISTRO_CMD_FOR_INSTALL_PACKAGE` or something similar.

Reply:
okay